site/pages/api/hootsuite-callback.api.ts

```diff
@@ -0,0 +1,110 @@
+import * as jose from "jose";
+import { NextApiRequest, NextApiResponse } from "next";
+import {
+    COOKIES_ID_TOKEN,
+    COOKIES_REFRESH_TOKEN,
+    HOOTSUITE_URL,
+    SOCIAL_MEDIA_ACCOUNTS_PAGE_PATH,
+} from "../../constants";
+import { getParameter, putParameter } from "../../data/ssm";
+import { initiateRefreshAuth } from "../../middleware.api";
+import { hootsuiteMeSchema, hootsuiteTokenSchema } from "../../schemas/hootsuite.schema";
+import { getSession } from "../../utils/apiUtils/auth";
+import { redirectToError, redirectTo, setCookieOnResponseObject } from "../../utils/apiUtils/index";
+
+const hootsuiteCallback = async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        const { code, state } = req.query;
+
+        const idToken = state?.toString() ? await getParameter(`/${state.toString()}/token`) : undefined;
+        const refreshToken = state?.toString() ? await getParameter(`/${state.toString()}/refresh-token`) : undefined;
+
+        const clientId = await getParameter(`/social/hootsuite/client_id`);
+        const clientSecret = await getParameter(`/social/hootsuite/client_secret`);
```

Comment:
Wrap in Promise.all for simultaneous processing

Reply:
done